src/libpriv/rpmostree-core.c

```diff
@@ -1715,6 +1739,88 @@ checkout_package_into_root (RpmOstreeContext *self,
   return TRUE;
 }
 
+static Header
+get_rpmdb_pkg_header (rpmts rpmdb_ts,
+                      DnfPackage *pkg,
+                      GCancellable *cancellable,
+                      GError **error)
+{
+  g_auto(rpmts) rpmdb_ts_owned = NULL;
+  if (!rpmdb_ts) /* allow callers to pass NULL */
+    rpmdb_ts = rpmdb_ts_owned = rpmtsCreate ();
+
+  unsigned int dbid = dnf_package_get_rpmdbid (pkg);
+  g_assert (dbid > 0);
+
+  g_auto(rpmdbMatchIterator) it =
+    rpmtsInitIterator (rpmdb_ts, RPMDBI_PACKAGES, &dbid, sizeof(dbid));
+
+  g_auto(Header) hdr = it ? rpmdbNextIterator (it) : NULL;
+  if (hdr == NULL)
+    return glnx_null_throw (error, "Failed to find package '%s' in rpmdb",
+                            dnf_package_get_nevra (pkg));
+
+  return headerLink (g_steal_pointer (&hdr));
```

Comment:
Do we need `headerLink`?  Aren't we returning our ref to it?  Either that or the `g_auto()` on the `Header` is wrong, correct?